### PR TITLE
Add Bukkit's new `provides` field.

### LINKED ISSUE #6: Add support for provides option for Bukkit
https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/commits/9f2fd967a4f914b849afe6ac287e7a595c035884#src/main/java/org/bukkit/plugin/PluginDescriptionFile.java, nuff said

### PR DESCRIPTION
**Pull request by Proximyst**
Fixes #6

Adds the new `provides` field for Bukkit along with verification for it.

Commit adding this field: https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/commits/9f2fd967a4f914b849afe6ac287e7a595c035884#src/main/java/org/bukkit/plugin/PluginDescriptionFile.java

### CODE REVIEW

**Comment by stephan-gh** [comment]
This looks like Spigot's (not Bukkit's) new `provides` field :P Bukkit is dead!

**Comment by stephan-gh** [comment]
Can you add an example to the README?

File: src/main/kotlin/net/minecrell/pluginyml/bukkit/BukkitPlugin.kt [hunk 1]
```diff
@@ -60,6 +60,10 @@ class BukkitPlugin : PlatformPlugin<BukkitPluginDescription>("Bukkit", "plugin.y
                 if (alias.contains(':')) throw InvalidPluginDescriptionException("Alias '$alias' of '${command.name}' cannot contain ':'")
             }
         }
+
+        if (description.provides?.all(VALID_NAME::matches)?.not() == true) {
```
Comment (stephan-gh) [inline src/main/kotlin/net/minecrell/pluginyml/bukkit/BukkitPlugin.kt]: What is the `?.not()` here? Why not just

```suggestion
        if (description.provides?.all(VALID_NAME::matches) == false) {
```

**Review by stephan-gh — APPROVED** [review]

**Comment by mikroskeem** [comment]
Any updates?

**Pull request merged**